{{cookiecutter.app_name}}/bin/check-coverage.sh

```diff
@@ -0,0 +1,20 @@
+#!/bin/sh
+
+# go tool cover -func coverage/coverage.out results in the multiline text telling coverage percentage on each function in the following format
+#   github.com/nimblehq/gulf-approval-web/helpers/config.go:8:                      GetConfigPrefix    100.0%
+#   github.com/nimblehq/gulf-approval-web/helpers/config.go:28:                     GetStringConfig    100.0%
+#   github.com/nimblehq/gulf-approval-web/lib/api/v1/controllers/health.go:13:      HealthStatus       100.0%
+#   total:                                                                          (statements)       100.0%
+# grep total to get the line start with `total` which contain the overall coverage percentage
+# awk '{print substr($3, 1, length($3)-1)}' with the built-in variable `$3` to grab the 3rd part after the line is splited with space and substr to remove the %
+# which will result in 100.0
+coverage=$(go tool cover -func coverage/coverage.out | grep total | awk '{print substr($3, 1, length($3)-1)}')
+expected=100
+
+if [ "$(echo $coverage '<' $expected | bc -l)" -eq 1 ]; then\
+  echo "coverage percentage is too low ${coverage}%, the expected percentage is ${expected}%"
```

Comment:
Should we titleize the sentence, and also add color to the text? Then we can see the success or warning text 😄 
https://stackoverflow.com/questions/5947742/how-to-change-the-output-color-of-echo-in-linux

Reply:
Nice, updated in [017d1c8](https://github.com/nimblehq/gin-templates/pull/95/commits/017d1c8f6dac32c0e7efe9f9543aa6839f2086a6)

![Screenshot 2566-03-21 at 14 45 26](https://user-images.githubusercontent.com/1772999/226548733-e2430529-44b5-4da2-b3ab-180f83dd7758.png)
![Screenshot 2566-03-21 at 14 45 53](https://user-images.githubusercontent.com/1772999/226548736-0254ad8d-20a0-4b4d-9fde-61360bdefcc8.png)